float32 Rx
float32 Ry
float32 Rz
float32 Ri